lib/highlighted-area-view.coffee

```diff
@@ -110,28 +123,38 @@ class HighlightedAreaView
         regexSearch =  "\\b" + regexSearch
       regexSearch = regexSearch + "\\b"
 
-    resultCount = 0
+    localCount = 0
     editor.scanInBufferRange new RegExp(regexSearch, regexFlags), range,
-      (result) =>
-        resultCount += 1
-        unless @showHighlightOnSelectedWord(result.range, @selections)
-          marker = editor.markBufferRange(result.range)
-          decoration = editor.decorateMarker(marker,
-            {type: 'highlight', class: @makeClasses()})
-          @views.push marker
-          @emitter.emit 'did-add-marker', marker
-
-    @statusBarElement?.updateCount(resultCount)
-
-  makeClasses: ->
-    className = 'highlight-selected'
+    (result) =>
+      localCount += 1
+      unless @showHighlightOnSelectedWord(result.range, @selections)
+        marker = editor.markBufferRange(result.range)
+        decoration = editor.decorateMarker(marker,
+          {type: 'highlight', class: @makeClasses(i+1)})
+        @views.push marker
+        @emitter.emit 'did-add-marker', marker
+
+    @selectedWords.push text
+    @selectedCounts.push localCount
+
+    # pass in an array of selected words and their counts in respect to update
+    # status count view.
+    @statusBarElement?.updateCount(@selectedWords, @selectedCounts)
+
+  #There are multiple css classes that individually have an rgb code
+  #Calls the specific css class using number, can add more colors and themes by
+  #adding more css classes in the highlight-selected.less file
+  makeClasses: (number) ->
+    className = 'highlight-selected selection'+number
     if atom.config.get('highlight-selected.lightTheme')
```

Comment:
For your package, this config should be updated to `atom.config.get('multi-highlight-selected.lightTheme)`. 

For merging in it is ok.